### PR TITLE
Use packages for the common RESTEasy reflection ignores and add the JSON packages to DEFAULT_IGNORED_PACKAGES

### LINKED ISSUE #11208: JsonNode not in Jandex after upgrade to 1.7.0.CR1
**Describe the bug**
In 1.6.1.Final the maven build don't show any warning, but changing the version to 1.7.0.CR1 triggers a WARNING about `com.fasterxml.jackson.databind.JsonNode` that is not in the index.

Is a multi-module project using RESTEasy, so I would assume there is the ignore for ReflectiveHierarchyStep set.

**Expected behavior**
No log warning message.

**Actual behavior**
```
[WARNING] [io.quarkus.deployment.steps.ReflectiveHierarchyStep] Unable to properly register the hierarchy of the following classes for reflection as they are not in the Jandex index:
	- com.fasterxml.jackson.databind.JsonNode (source: io.fabric8.kubernetes.api.model.HasMetadata)
Consider adding them to the index either by creating a Jandex index for your dependency via the Maven plugin, an empty META-INF/beans.xml or quarkus.index-dependency properties.");.
```
**To Reproduce**
** Not sure how to reproduce it easily **

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: Linux qk 5.4.0-42-generic #46-Ubuntu SMP Fri Jul 10 00:24:02 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
 - Output of `java -version`: OpenJDK Runtime Environment GraalVM CE 20.1.0 (build 11.0.7+10-jvmci-20.1-b02)
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 1.7.0.CR1
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`):  3.6.3

### PR DESCRIPTION
**Pull request by gsmet**
We used to add classes one at a time but in all those cases, adding the
whole package makes more sense as everything will be handled by RESTEasy
and specific (de)serializers.

I was sure I had created a PR for that branch but apparently missed it...

Fixes #11208

### CODE REVIEW

File: extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyDotNames.java [hunk 1]
```diff
@@ -80,28 +80,19 @@ public boolean test(DotName dotName) {
...
 
+    private static final String[] PACKAGES_IGNORED_FOR_REFLECTION = {
+            // JSON-P
+            "javax.json.",
+            // Jackson
+            "com.fasterxml.jackson.databind.",
             // JAX-RS
-            DotName.createSimple("javax.ws.rs.core.Response"),
-            DotName.createSimple("javax.ws.rs.core.Response.StatusType"),
-            DotName.createSimple("javax.ws.rs.container.AsyncResponse"),
```
Comment (jorsol) [inline extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyDotNames.java:89]: This one is missing, it's on the `javax.ws.rs.container` package, not on `javax.ws.rs.core`

Comment (gsmet) [reply extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyDotNames.java:89]: Good catch, fixed!

**Review by gastaldi — APPROVED** [review]

File: extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyDotNames.java [hunk 2]
```diff
@@ -80,28 +80,19 @@ public boolean test(DotName dotName) {
 
     // Types ignored for reflection used by the RESTEasy and SmallRye REST client extensions.
     private static final Set<DotName> TYPES_IGNORED_FOR_REFLECTION = new HashSet<>(Arrays.asList(
-            // Jackson
-            DotName.createSimple("com.fasterxml.jackson.databind.JsonNode"),
+    // Consider adding packages below instead if it makes more sense
+    ));
 
+    private static final String[] PACKAGES_IGNORED_FOR_REFLECTION = {
```
Comment (machi1990) [inline extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyDotNames.java:86]: Wondering if we could provide a config item for user to build this list up by themselves? Of course we can have good defaults, but we have been adding classes/ packages on a continual basis here.

Comment (gsmet) [reply extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyDotNames.java:86]: Yeah, so I kinda agree. What makes me hesitant is that I fear they won't report these things anymore, thus we would have a bad experience for new users.

Comment (machi1990) [reply extensions/resteasy-common/deployment/src/main/java/io/quarkus/resteasy/common/deployment/ResteasyDotNames.java:86]: Yes, but new users will ran into these warnings and report them anyway. We could either add the configuration key to the warning message to make the experience even better or once the issue is reported see if we need to review the "default values" or something more refined need to be done.

**Pull request merged**